src/main/java/ServiceLayer/ShopService.java

```diff
@@ -26,13 +28,49 @@ public ShopService(){
     * @param userName  The name of the user opening the shop (founder).
     * @return          A response indicating the success or failure of the operation.
     */
-    public Response OpenNewShop(Integer shopId, String userName)
+    public Response openNewShop(String token,Integer shopId, String userName)
     {
         Response response = new Response();
         try
         {
-            _shopController.OpenNewShop(shopId, userName);
-            logger.info(String.format("New shop created by: %s with Shop ID: %d" ,userName, shopId));
+            if (_tokenService.validateToken(token)) 
```

Comment:
think about- if-else if-else istead of if-if-else-else

Reply:
if (!_tokenService.validateToken(token)) 
          throw new Exception("Invalid session token.");
if......